### ISSUE TITLE
Kernel dies immediately in fresh docker image; bad interface file

### ISSUE THREAD

**Issue by cemulate** [gibiansky/IHaskell#933]
After cloning the repo and running

    $ docker build -t ihaskell:latest .
    $ docker run -it -p8888:8888 ihaskell:latest

directly from the documentation, I logged in with the provided token and created a new "Haskell" notebook, the kernel died immediately. Here are the logs:

```
Kernel started: 81fd90cc-1290-45bc-9697-b1d079f8b27e
Writing implicit global project config file to: /home/jovyan/.stack/global-project/stack.yaml
Using latest snapshot resolver: lts-12.9
Downloading lts-12.9 build plan ...
Downloaded lts-12.9 build plan.
Preparing to install GHC to an isolated location.
This will not interfere with any system-level installation.
Preparing to download ghc-8.4.3 ...
ghc-8.4.3: download has begun
ghc-8.4.3:    1.13 MiB / 144.83 MiB (  0.78%) downloaded...
ghc-8.4.3:    3.24 MiB / 144.83 MiB (  2.23%) downloaded...
...
ghc-8.4.3:  143.29 MiB / 144.83 MiB ( 98.93%) downloaded...
ghc-8.4.3:  144.83 MiB / 144.83 MiB (100.00%) downloaded...
Downloaded ghc-8.4.3.
Unpacking GHC into /home/jovyan/.stack/programs/x86_64-linux/ghc-8.4.3.temp/ ...
Configuring GHC ...
Installing GHC ...
Installed GHC.
ihaskell: Bad interface file: /home/jovyan/.stack/programs/x86_64-linux/ghc-8.4.3/lib/ghc-8.4.3/base-4.11.1.0/Prelude.hi
    mismatched interface file versions (wanted "8022", got "8043")
Bad interface file: /home/jovyan/.stack/programs/x86_64-linux/ghc-8.4.3/lib/ghc-8.4.3/directory-1.3.1.5/System/Directory.hi
    mismatched interface file versions (wanted "8022", got "8043")
Bad interface file: /home/jovyan/.stack/programs/x86_64-linux/ghc-8.4.3/lib/ghc-8.4.3/unix-2.7.2.2/System/Posix/IO.hi
    mismatched interface file versions (wanted "8022", got "8043")
Bad interface file: /home/jovyan/.stack/programs/x86_64-linux/ghc-8.4.3/lib/ghc-8.4.3/base-4.11.1.0/System/IO.hi
    mismatched interface file versions (wanted "8022", got "8043")
Bad interface file: /home/jovyan/.stack/programs/x86_64-linux/ghc-8.4.3/lib/ghc-8.4.3/template-haskell-2.13.0.0/Language/Haskell/TH.hi
    mismatched interface file versions (wanted "8022", got "8043")
```

followed by the `KernelRestarter` attempting to restart it with the same failure every time. 

Am I supposed to be cloning the repo and building the docker image from there? Or is there a supported image on dockerhub that I should be using? I saw [this](https://hub.docker.com/r/gibiansky/ihaskell/), but it was last pushed 3 years ago so I'm not sure if it's supported.

**Comment by vaibhavsagar** [comment]
This is because the resolver in `stack.yaml` doesn't match the latest version. Can you try doing this:

```bash
$ mv stack-8.4.yaml stack.yaml
```

and rebuilding?

**Comment by cemulate** [comment]
I tried that, but then the build fails with on 32/36 with:

```
Step 32/36 : RUN stack build && stack install
 ---> Running in 98aef1ca3963
ghc-parser-0.2.0.2: configure (lib)
magic-1.1: download
Chart-1.9: download
ghc-parser-0.2.0.2: build (lib)
ipython-kernel-0.10.0.0: configure (lib)
ipython-kernel-0.10.0.0: build (lib)
...
ihaskell-widgets-0.2.3.2: build (lib)
ihaskell-static-canvas-0.1.0.0: copy/register
stack: internal error: Unable to commit 1586495488 bytes of memory
    (GHC version 8.2.2 for x86_64_unknown_linux)
    Please report this as a GHC bug:  http://www.haskell.org/ghc/reportabug
Aborted (core dumped)
The command '/bin/sh -c stack build && stack install' returned a non-zero code: 134
```

which certainly seems odd. I don't know much about Haskell ecosystems / deployment, but there's also a `release-8.*.nix` in the directory, is it possible I need to `mv release-8.4.nix release.nix`?

**Comment by cemulate** [comment]
I commented out the two lines about `ihaskell-widgets` (under `packages` and `ghc-options`) and it worked.

**Comment by vaibhavsagar** [comment]
I think the issue is that `ihaskell-widgets` is a large module and you ran into memory limitations compiling it.

**Comment by cemulate** [comment]
Interesting. I certainly have more than 1.5G available - not sure if I need to tell `docker build` to use more memory or `stack`; any ideas?

**Comment by vaibhavsagar** [comment]
No, if you have enough memory I would expect it to work. What happens if you re-enable `ihaskell-widgets` and rebuild?

**Comment by cemulate** [comment]
I haven't gotten back trying to fix this; I ended up building off one of my own docker images and installing IHaskell there, which worked just fine. If I get a chance though I'll continue seeing if I can make this Dockerfile build sucessfully. Feel free to close, however, if nobody else is having issues.

**Comment by mildlyincompetent** [comment]
I was having the same error of `mismatched interface file versions (wanted "8022", got "8043")` in docker and on my own machine, switching to `stack-8.4.yaml` resolved the issue.

As for build memory requirements, it appears the build is rather memory intensive, but the optimization seems to be even more memory intensive. My build was using around 12G of memory at the end.

You may want to try running `stack` with the `--fast` option which I believe skips the optimization steps and seems likely to significantly decrease memory requirements.

**Comment by vaibhavsagar** [comment]
Closed by #952.
